### PR TITLE
Melee attacks no longer lock inputs

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Makes it so attacking no longer checks next_move, but rather checks last_attack
## Why It's Good For The Game
Melee input locking is cringe and should go away
## Changelog
:cl:
qol: melee attacking no longer locks your inputs
/:cl:

### CODE REVIEW

**Comment by Lauwma** [comment]
Machete finally useful ?

**Comment by chizzy376** [comment]
vali claymore buff of the millennia

File: code/_onclick/ai.dm [hunk 1]
```diff
@@ -32,9 +32,9 @@
 
 
 /mob/living/silicon/ai/ClickOn(atom/A, location, params)
-	if(world.time <= next_click)
+	if(world.time <= last_click)
 		return
-	next_click = world.time + 1
+	last_click = world.time + 1
```
Comment (QualityVan) [inline code/_onclick/ai.dm:37]: Setting a var called 'last click' to a point in the future is incredibly weird.

Comment (Xander3359) [reply code/_onclick/ai.dm:37]: This is effectively storing the time before you can click again.
Thus, to pass this check, world.time has to be > than last_click.

What do you think would be a more appropriate name if not this?

Comment (MLGTASTICa) [reply code/_onclick/ai.dm:37]: next_click is more appropiate , it stores the time at which the next click can happen

**Review by Kuro020 — APPROVED** [review]

File: code/_onclick/ai.dm (hunk 1 above)
Comment (Xander3359) [reply code/_onclick/ai.dm]: Sure that makes sense now.

**Review by TiviPlus — APPROVED** [review]

File: code/_onclick/item_attack.dm [hunk 2]
```diff
@@ -163,23 +160,22 @@
 
 	return TRUE
 
-
 /mob/living/attackby(obj/item/I, mob/living/user, params)
 	. = ..()
 	if(.)
 		return TRUE
-	user.changeNext_move(I.attack_speed)
+	user.last_attack = world.time + I.attack_speed
```
Comment (TiviPlus) [inline code/_onclick/item_attack.dm:167]: shoudlnt this be named next_attack

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Tming to test

**Review by Lumipharon — CHANGES_REQUESTED** [review]
This has some issues.
The CD is only applied on MOB attackby, so excludes alt attacks (i.e. right click), punch attacks, and attacking anything that's not a mob.

**Comment by Looking4bros** [comment]
F

**Comment by Zangeiti** [comment]
F

**Pull request closed without merging**